### PR TITLE
Master onboarding - include "Basics" module cards in the newcomer's Learning Path

### PR DESCRIPTION
**Pull request by AdriAt360**
https://github.com/360Learning/360learning.github.io/issues/6

## Context
The magic begins =)
Here is a screenshot of the improved tool
image

And here is a video of the result of including the basics
https://www.loom.com/share/6e7c1919f8ec42b087fd6eb0f0fd4159

As I wrote in a comment, I intentionally hardcoded a fake suffix to prevent me from testing in production 🤣 

## Changes
The idea is to fetch all the cards that need to be included, and then to include them (by posting a comment on them)

Here as well, you can see that I don't use any framework, because the code is still simple enough. I feel the whole tool will be feasible without too much complexity here.

### CODE REVIEW

File: scripts/trello/api.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+async function put(relativeUrl, params, { trelloApiKey, trelloOAuth1 } = {}) {
+    const res = await axios.put(buildUrlWithParams(trelloApiKey, trelloOAuth1, relativeUrl, params));
+    return res.data;
+}
+
+function buildUrlWithParams(trelloApiKey, trelloOAuth1, relativeUrl, params) {
+    const rawUrl = `https://api.trello.com/1/${relativeUrl}?key=${trelloApiKey}&token=${trelloOAuth1}`;
+    return params ? `${rawUrl}&${encodeURI(params)}` : rawUrl;
```
Comment (pp0rtal) [inline scripts/trello/api.js]: @AdriAt360 If you put everything in a single object, will emove the ternary, the manual param concatenation
```javascript
const urlParams = { ...params, key, token };
return `${rawUrl}?${encodeURI(urlParams)}`
```

File: scripts/trello/cards.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+async function postComment(cardId, text, { trelloApiKey, trelloOAuth1 } = {}) {
+    return post(
+        `cards/${cardId}/actions/comments`,
+        `text=${text}`,
```
Comment (pp0rtal) [inline scripts/trello/cards.js]: @AdriAt360 Could be more maintainable to pass `{ text }` in here and let your client lib do the encoding

File: scripts/trello/api.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+async function get(relativeUrl, params, { trelloApiKey, trelloOAuth1 } = {}) {
```
Comment (pp0rtal) [inline scripts/trello/api.js]: @AdriAt360 get / post / put are becoming globals. I know the project will keep small, but it's possibly already too dirty.

 - Could be interesting to at least "tidy" them in a TrelloApi to use them like `TrelloClient.get(...)`.
 - Other idea: keep key / token in an instance contex like this, which will simplify the context handling in your project
```javascript
const trelloClient = new TrelloClient(key, token);
trelloClient.get(** less_infos here :) **);
```

File: scripts/trello/api.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    const res = await axios.post(buildUrlWithParams(trelloApiKey, trelloOAuth1, relativeUrl, params));
+    return res.data;
+}
+
+async function put(relativeUrl, params, { trelloApiKey, trelloOAuth1 } = {}) {
+    const res = await axios.put(buildUrlWithParams(trelloApiKey, trelloOAuth1, relativeUrl, params));
+    return res.data;
+}
+
+function buildUrlWithParams(trelloApiKey, trelloOAuth1, relativeUrl, params) {
```
Comment (pp0rtal) [inline scripts/trello/api.js:22]: @AdriAt360 Old school tip if you want to create package without a bundler:
you can wrap it in a IIFE
```javascript
const { exposedElement } = (() => {
    ... private definitions
    return { exposedElement };
})()` // <- fork bomb style
```

Will make the whole file cleaner, and will hide this method from global scope.

**Review by pp0rtal — APPROVED** [review]
@AdriAt360 Overall I'm a bit disappointed you didn't set up a Typescript front-end stack
_(obviously it's because this project is not important and already nearly over)_

Just a few suggestion about packaging / contextualize, optional.

**Comment by AdriAt360** [comment]
@pp0rtal 
> Overall I'm a bit disappointed you didn't set up a Typescript front-end stack

^^
the main problem here is that it's Github Pages, that are static site hosting service that deploys static apps straight from the repository, so everything implying packaging/frameworks/transpiling is cumbersome ><

I'll look at your other comments, they are interesting 👍🏽

**Review by antoine-roth — APPROVED** [review]
@AdriAt360 all good 👍

File: scripts/trello/api.js (hunk 4 above)
Comment (AdriAt360) [reply scripts/trello/api.js:22]: @pp0rtal I'm not sure I will do that yet, for readability's sake in this file (ok, I'll keep these functions global, but there are no consequences), but maybe in the next PR
when you say "create package", do you mean that I would be able to `import` directly stuff in the other files? Seems tough as the `import` must be done inside a module, but if so, I might do that in the next ticket

File: scripts/trello/api.js (hunk 1 above)
Comment (AdriAt360) [reply scripts/trello/api.js]: @pp0rtal nice suggestion, done 👍🏽 https://github.com/360Learning/360learning.github.io/pull/10/commits/2ff73931a0eddb75168722cabf0a22f1a3046006

File: scripts/trello/cards.js (hunk 2 above)
Comment (AdriAt360) [reply scripts/trello/cards.js]: @pp0rtal yes sir! https://github.com/360Learning/360learning.github.io/pull/10/commits/0ed2235e2f538ed602527435750b7130cc312015

File: scripts/trello/api.js (hunk 3 above)
Comment (AdriAt360) [reply scripts/trello/api.js]: @pp0rtal good idea to introduce a `TrelloClient` here! https://github.com/360Learning/360learning.github.io/pull/10/commits/c480d98c0f9f102eb3e351e63eef76b513b7a61c

File: scripts/trello/api.js (hunk 4 above)
Comment (pp0rtal) [reply scripts/trello/api.js:22]: @AdriAt360 My suggestion was about exposing function which needed to be exposed. As here, any function is global. Wrapping them into an IIFE can make your private functions have the file scope.

I don't know about import/export on front-end side, I didn't code on front-end for years (or only on stacked project) :smile:

**Comment by pp0rtal** [comment]
> ^^
> the main problem here is that it's Github Pages, that are static site hosting service that deploys static apps straight from the repository, so everything implying packaging/frameworks/transpiling is cumbersome ><

@AdriAt360 Ok, so I remove my "I'm a bit disappointed". I'm sure you were hurt.

**Comment by AdriAt360** [comment]
@pp0rtal [this is madness!](https://www.youtube.com/watch?v=rvYZRskNV3w)

https://github.com/360Learning/360learning.github.io/pull/10/commits/d385c9f9bb6f619740cfa5848c8bdf937a05a5db

**Pull request merged**